crates/sui-graphql-rpc/src/server/version.rs

```diff
@@ -50,57 +49,49 @@ impl headers::Header for SuiRpcVersion {
 /// one version of the RPC software, and it is the responsibility of the load balancer to make sure
 /// version constraints are met.
 pub(crate) async fn check_version_middleware<B>(
-    user_version: Option<TypedHeader<SuiRpcVersion>>,
     State(version): State<Version>,
     request: Request<B>,
     next: Next<B>,
 ) -> Response {
-    if let Some(TypedHeader(SuiRpcVersion(req_version, rest))) = user_version {
-        if !rest.is_empty() {
-            return (
-                StatusCode::BAD_REQUEST,
-                graphql_error_response(
-                    code::BAD_REQUEST,
-                    format!("Failed to parse {VERSION_HEADER}: Multiple possible versions found."),
-                ),
-            )
-                .into_response();
-        }
-
-        let Ok(req_version) = std::str::from_utf8(&req_version) else {
-            return (
-                StatusCode::BAD_REQUEST,
-                graphql_error_response(
-                    code::BAD_REQUEST,
-                    format!("Failed to parse {VERSION_HEADER}: Not a UTF8 string."),
-                ),
-            )
-                .into_response();
-        };
-
-        let Some((year, month)) = parse_version(req_version) else {
-            return (
-                StatusCode::BAD_REQUEST,
-                graphql_error_response(
-                    code::BAD_REQUEST,
-                    format!(
-                        "Failed to parse {VERSION_HEADER}: '{req_version}' not a valid \
+    let route = request.uri().path();
+    let parts = route.split_once("/graphql");
+    if let Some((_, rest)) = parts {
+        if let Some(query_version) = rest.strip_prefix('/') {
```

Comment:
What happens if someone passes in `/graphql/`? It seems like `rest` would be `"/"` and `query_version` would be `""`, and then the version check would fail, when we should be treating that as if it were no version check at all.

Reply:
Can you add a test for this case?

Reply:
Actually, this returns a 404 from `curl`. This route wouldn't be accepted by `axum`, from my understanding. I checked if the `check_version_middleware` is being called on this route, and it's not. 
Added a test.